File: ecommerce-backend/Models/Product.cs
using System;
using System.Collections.Generic;

namespace ecommerce_backend.Models;

public partial class Product
{
    public int ProductId { get; set; }

    public int? CategoryId { get; set; }

    public int? SupplierId { get; set; }

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    public virtual ICollection<CampaignVariant> CampaignVariants { get; set; } = new List<CampaignVariant>();

    public virtual Category? Category { get; set; }

    public virtual ICollection<ProductReview> ProductReviews { get; set; } = new List<ProductReview>();

    public virtual Supplier? Supplier { get; set; }

    public virtual ICollection<Variant> Variants { get; set; } = new List<Variant>();

    public virtual ICollection<Attribute> Attributes { get; set; } = new List<Attribute>();
}

File: ecommerce_frontend/src/Models/Product.ts
import { CreateProuctAttributeDto } from "~/pages/admin/Product/FormProduct";
import { AttributeGet } from "./Attribute";
import { CategoryGet } from "./Category";
import { SupplierGet } from "./Supplier";
import { VariantGet } from "./Variant";

export type ProductGet = {
    productId: number;
    name: string;
    description: string;
    categoryDto: CategoryGet;
    supplierDto: SupplierGet;
    attributes: AttributeGet[];
    variants: VariantGet[]
}

export type ProductPost = {
    categoryId: number;
    supplierId: number;
    name: string;
    description: string;
    attributes: CreateProuctAttributeDto[];

}

export type ProductPut = {
    categoryId: number;
    supplierId: number;
    name: string;
    description: string;
}